backend/icpy/services/filesystem_service.py

```
@@ -911,6 +911,69 @@ async def write_file(self, file_path: str, content: str, encoding: str = 'utf-8'
+
+            # Publish event (buffered)
+            await self._publish_or_buffer('fs.file_written', {
+                'file_path': file_path,
+                'size': len(content),
+                'binary': True,
+                'created': not file_exists,
+                'timestamp': time.time()
+            })
+            
+            logger.info(f"Binary file written successfully: {file_path} ({len(content)} bytes)")
+            return True
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)